File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.java
class Solution {
    public int[] sortByBits(int[] arr) {
        List<int[]> res = new ArrayList<>();

        for(int val : arr){
            int bits = Integer.bitCount(val);
            res.add(new int[]{bits , val});
        }


        Collections.sort(res , (a , b) -> {
            if(a[0] != b[0]){
                return a[0] - b[0];    //sort on basis of bits
            }

            return a[1] - b[1];    //sort on basis of number.
        });

        for(int i = 0; i < arr.length; i++){
            arr[i] = res.get(i)[1];
        }

        return arr;
    }
}

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        vector<pair<int , int>> res;

        for(auto val : arr){
            int bits = __builtin_popcount(val);
            res.push_back({bits , val});
        }

        sort(res.begin() , res.end());

        arr.clear();

        for(auto val : res){
            arr.push_back(val.second);
        }

        return arr;
    }
};